third_party/ObjexxFCL/src/ObjexxFCL/Array.hh

```diff
@@ -1776,7 +1776,11 @@ public: // Comparison: Predicate: Any
 		assert( a.size_bounded() );
 		if ( a.empty() ) return false;
 		for ( size_type i = 0, e = a.size_; i < e; ++i ) {
-			if ( t < a[ i ] ) return true;
+			if (t < a[i]) {
```

Comment:
Hmm, we generally don't modify the third-party libraries directly. What's the reason for these changes? It would help if you could specifically outline your reasons for making these changes, such as pointing us to some code in E+ that's misbehaving so we can have some context.

Reply:
I concur, I hope that this isn't needed.  I think what we probably need to do instead is just get rid of the call to `any_lt`, and instead do this loop/check ourselves right in EnergyPlus code.  That way we don't have to edit this.  But please speak up if I'm missing something here.  Otherwise this looks good.

Reply:
Yes. Getting rid of `any_lt` should fix the errors happening now. Just that there is a (very) remote possibility of this happening again in the future if someone else tries to use `any_lt` for an array that was linearly interpolated like the case here.

Reply:
As context, for my test, [`any_gt`](https://github.com/NREL/EnergyPlus/blob/7a650ed251d6fa927d61df916e81504ab3a55f9d/src/EnergyPlus/ScheduleManager.cc#L947) returns false which causes the warning in [this line](https://github.com/NREL/EnergyPlus/blob/7a650ed251d6fa927d61df916e81504ab3a55f9d/src/EnergyPlus/ScheduleManager.cc#L948). 

[`any_gt`](https://github.com/NREL/EnergyPlus/blob/7a650ed251d6fa927d61df916e81504ab3a55f9d/third_party/ObjexxFCL/src/ObjexxFCL/Array.hh#L1813) calls [`any_lt`](https://github.com/NREL/EnergyPlus/blob/7a650ed251d6fa927d61df916e81504ab3a55f9d/third_party/ObjexxFCL/src/ObjexxFCL/Array.hh#L1761) where the comparisons are done. 

I guess there are two `any_lt` functions as well ([here ](https://github.com/NREL/EnergyPlus/blob/7a650ed251d6fa927d61df916e81504ab3a55f9d/third_party/ObjexxFCL/src/ObjexxFCL/Array.hh#L1761) and [here](https://github.com/NREL/EnergyPlus/blob/7a650ed251d6fa927d61df916e81504ab3a55f9d/third_party/ObjexxFCL/src/ObjexxFCL/Array.hh#L1774)), and it's necessary to change both if we decide to change one. 

There's already a check in [this line](https://github.com/NREL/EnergyPlus/blob/7a650ed251d6fa927d61df916e81504ab3a55f9d/src/EnergyPlus/ScheduleManager.cc#L2278). I'm looking if this is enough to remove the other check.

Reply:
I removed these changes because the limits are being validated at a later part of the code, [here](https://github.com/NREL/EnergyPlus/blob/f84f89f6d6cb935885ca7af3fea3664fe6f6516b/src/EnergyPlus/ScheduleManager.cc#L2271).